src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderSettingsDao.java

```diff
@@ -0,0 +1,86 @@
+package bio.terra.service.snapshotbuilder;
+
+import bio.terra.model.SnapshotBuilderSettings;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.UUID;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.ServerErrorException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+public class SnapshotBuilderSettingsDao {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+  private static ObjectMapper objectMapper = new ObjectMapper();
+
+  private static class SnapshotBuilderSettingsMapper implements RowMapper<SnapshotBuilderSettings> {
+
+    public SnapshotBuilderSettings mapRow(ResultSet rs, int rowNum) throws SQLException {
+
+      try {
+        return objectMapper.readValue(rs.getString("settings"), SnapshotBuilderSettings.class);
+      } catch (JsonProcessingException e) {
+        throw new ServerErrorException("Settings not stored as properly formatted json", 500, e);
+      }
+    }
+  }
+
+  @Autowired
+  public SnapshotBuilderSettingsDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+  public SnapshotBuilderSettings getSnapshotBuilderSettingsByDatasetId(UUID datasetId) {
+    try {
+
+      return jdbcTemplate.queryForObject(
+          "SELECT settings FROM snapshot_builder_settings WHERE dataset_id = :dataset_id",
+          new MapSqlParameterSource().addValue("dataset_id", datasetId),
```

Comment:
Personally I would use `Map.of()` here. Although it looks like most TDR code uses `MapSqlParameters`.. I think `Map.of()` is just as clear and is less code.
```suggestion
          Map.of("dataset_id", datasetId),
```